src/SourceTokenList.ts

```diff
@@ -37,7 +35,7 @@ export default class SourceTokenList {
    * Map each token to an element of an array.
    */
   map<T>(mapper: (token: SourceToken, index: SourceTokenListIndex, list: SourceTokenList) => T): Array<T> {
-    let result = [];
+    let result: Array<T> = [];
```

Comment:
This was the most common sort of change. TypeScript doesn't infer the type of the array based on usage, so you have to tell it.